File: ArtSiteWeb/artsite-web/src/network/models/characterModel.ts
import { Gender } from "../enums/gender";
import { Sexuality } from "../enums/sexuality";
import { AppearanceModel } from "./appearanceModel";
import { ArtistModel } from "./artistModel";
import { ArtworkModel } from "./artworkModel";
import { BaseNameModel } from "./baseNameModel";
import { GeneralInfoModel } from "./generalInfoModel";
import { InterestsModel } from "./interestsModel";
import { PersonalityModel } from "./personalityModel";
import { ReferenceModel } from "./referenceModel";
import { SpeciesModel } from "./speciesModel";
import { TagModel } from "./tagModel";

export interface CharacterModel extends BaseNameModel {
    age: number;
    gender: Gender;
    sexuality: Sexuality;
    iconId: number;
    icon: ArtworkModel;
    tags: TagModel[];
    species: SpeciesModel;
    originalDesigner: ArtistModel;
    references: ReferenceModel[];
    generalInfo: GeneralInfoModel;
    personality: PersonalityModel;
    interests: InterestsModel;
    appearance: AppearanceModel;
}

File: ArtSiteApi/ArtSite.Data/Models/CharacterModel.cs
using ArtSite.Data.Enums;
using ReactiveUI;

namespace ArtSite.Data.Models; 

public class CharacterModel : BaseNameModel {
    public byte Age { get; set; }
    public Gender Gender { get; set; }
    public Sexuality Sexuality { get; set; }
    public int? IconId { get; set; }
    public string WantedArtwork { get; set; }
    public ArtworkModel? Icon { get; set; }
    public HashSet<TagModel> Tags { get; set; }
    public SpeciesModel Species { get; set; }
    public ArtistModel OriginalDesigner { get; set; }
    public HashSet<ReferenceModel> References { get; set; }
    public GeneralInfoModel GeneralInfo { get; set; }
    public PersonalityModel Personality { get; set; }
    public InterestsModel Interests { get; set; }
    public AppearanceModel Appearance { get; set; }
    
    public override string ToString() => Name;
}